lib/ops/ops.go

```diff
@@ -855,6 +855,143 @@ type Operations interface {
 	GetOperationPlan(SiteOperationKey) (*storage.OperationPlan, error)
 }
 
+// OperationsFilter represents a filter to apply to results when listing operations
+type OperationsFilter struct {
+	// Last indicates to only return the last operation
+	Last bool
+
+	// First indicates to only return the first operation
+	First bool
+
+	// Complete indicates to only return completed operations
+	Complete bool
+
+	// Finished indicates to only return finished operations (complete or failed)
+	Finished bool
+
+	// Active indicate to only return active operations
+	Active bool
+
+	// Type indicates to only return a type of operation
+	Type string
+}
+
+// URLValues converts the filter to a set of URL values that can be passed via the API
+func (f OperationsFilter) URLValues() (res url.Values) {
+	res = url.Values{}
+
+	if f.Last {
+		res.Add("last", "")
+	}
+
+	if f.First {
+		res.Add("first", "")
+	}
+
+	if f.Complete {
+		res.Add("complete", "")
+	}
+
+	if f.Finished {
+		res.Add("finished", "")
+	}
+
+	if f.Active {
+		res.Add("active", "")
+	}
+
+	if f.Type != "" {
+		res.Add("type", f.Type)
+	}
+
+	return
+}
+
+// FilterFromURLValues returns an operations filter based on set URL values
+func FilterFromURLValues(v url.Values) (f OperationsFilter) {
+	if _, ok := v["last"]; ok {
+		f.Last = true
+	}
+
+	if _, ok := v["first"]; ok {
+		f.First = true
+	}
+
+	if _, ok := v["complete"]; ok {
+		f.Complete = true
+	}
+
+	if _, ok := v["finished"]; ok {
+		f.Finished = true
+	}
+
+	if _, ok := v["active"]; ok {
+		f.Active = true
+	}
+
+	if t, ok := v["type"]; ok {
+		if len(t) > 0 {
+			f.Type = t[0]
+		}
+	}
+
+	return
+}
+
+// Filter takes a list of operations and filters the results based on the set filter parameters
+func (filter OperationsFilter) Filter(in SiteOperations) SiteOperations {
```

Comment:
A unit test for this wouldn't hurt.

Reply:
I can add some. I didn't bother since the operation system unit tests do largely cover this filtering through the util methods.